File: jan/adventofcode/2022/5/app2.py
stack1 = ["H","R","B","D","Z","F","L","S"]
stack2 = ["T","B","M","Z","R"]
stack3 = ["Z","L","C","H","N","S"]
stack4 = ["S","C","F","J"]
stack5 = ["P","G","H","W","R","Z","B"]
stack6 = ["V","J","Z","G","D","N","M","T"]
stack7 = ["G","L","N","W","F","S","P","Q"]
stack8 = ["M","Z","R"]
stack9 = ["M","C","L","G","V","R","T"]
stacks = [stack1, stack2, stack3, stack4, stack5, stack6, stack7, stack8, stack9]

total_score = ""

import re

def containter_printer(stacks):
    # find longest stack
    heighest_stack_height = 0
    for i in range(len(stacks)):
        heighest_stack_height = max(heighest_stack_height, len(stacks[i]))
    heighest_stack = max(range(len(stacks)), key=lambda i: len(stacks[i])) + 1

    # print from top to bottom
    for i in range(heighest_stack_height, -1, -1):
        row = ""
        for stack in stacks:
            if i > len(stack) - 1:
                row += "   "
            else:
                row += "[" + stack[i] + "]"
        print(row)
        if i == 0:
            print("[1][2][3][4][5][6][7][8][9]")

def container_mover(amount, from_stack, to_stack):
    # get amount from the end of the from_stack
    to_move = stacks[from_stack][-amount:]
    new_amount = len(to_move)

    # overwrite to_stack with append to end 
    to_stack = stacks[to_stack] + to_move
    stacks[to_stack] = to_stack

    # remove amount from from_stack and save to array
    from_stack = stacks[from_stack][:-new_amount]
    stacks[from_stack] = from_stack

# read input.txt line by line
with open("input.txt") as file:
    for line in file:
        print("\033[H\033[J")  # clear console
        containter_printer(stacks)
        print(line)
        input_ = [int(x) for x in re.findall(r"\b\d+\b", line)]
        container_mover(input_[0], input_[1] - 1, input_[2] - 1)

# do something after final line
containter_printer(stacks)
for stack in stacks:
    if len(stack) == 0:
        total_score += " "
    else:
        total_score += stack[-1]
print(f"the total is {total_score}")

File: jan/adventofcode/2022/5/app2.js
let stack1 = ["H","R","B","D","Z","F","L","S"]
let stack2 = ["T","B","M","Z","R"]
let stack3 = ["Z","L","C","H","N","S"]
let stack4 = ["S","C","F","J"]
let stack5 = ["P","G","H","W","R","Z","B"]
let stack6 = ["V","J","Z","G","D","N","M","T"]
let stack7 = ["G","L","N","W","F","S","P","Q"]
let stack8 = ["M","Z","R"]
let stack9 = ["M","C","L","G","V","R","T"]
let stacks = [stack1, stack2, stack3, stack4, stack5, stack6, stack7, stack8, stack9];

let totalScore = "";

//import txt file
var fs = require('fs'),readline = require('readline');
const { fileURLToPath } = require('url')
var rd = readline.createInterface({input: fs.createReadStream('input.txt'),console: false});

//functions
function containterPrinter(stacks){
    //find longest stack
    let heighestStackHeight = 0;
    for (let i = 0; i < stacks.length; i++) {heighestStackHeight = Math.max(heighestStackHeight, stacks[i].length);};

    //print from top to bottom
    for (var i = heighestStackHeight; i >= 0; i--) {
        let row ="";
        stacks.forEach(stack => {
            if(typeof stack[i] == 'undefined'){row = row.concat("   ");
            } else if(stack.length >= i){row = row.concat("[").concat(stack[i]).concat("]");
            } else {row = row.concat("   ")};
         });
         console.log(row);
         if(i === 0){console.log("[1][2][3][4][5][6][7][8][9]");}
    };
}

function containerMover(amount,from,to){
    //get from list and get amount from the end
    let toMove = stacks[from].slice((amount*-1));
    let newAmount = toMove.length;

    //overwrite to list with append to end 
    let toStack = stacks[to].concat(toMove);
    stacks[to] = toStack;

    //remove amount from from list and save to array
    let fromStack = stacks[from].slice(0, (stacks[from].length - newAmount));
    stacks[from] = fromStack;

}

//do something line by line of the imported file
rd.on('line', function(line) {
    containterPrinter(stacks);
    console.log(line);
    const input = line.match(/\b\d+\b/g);
    containerMover(input[0],input[1]-1,input[2]-1);
});

//do something after final line
rd.on('close', function() {
    containterPrinter(stacks);
    stacks.forEach(stack => {
        if(typeof stack[stack.length-1] == 'undefined'){totalScore = totalScore.concat(" ");}
        else {totalScore = totalScore.concat(stack[stack.length-1]);}
     });
    console.log(`the total is ${totalScore}`);
});